src/lib/atoms/Rating/Rating.tsx

```diff
@@ -0,0 +1,321 @@
+        setRegardingPosition(regradingPosition);
+        if (disableMouseMovie) return;
+
+        setHoveredValue(currentRating);
+        setRating(currentRating);
+    };
+
+    useLayoutEffect(() => {
+        if (isControlled || isDefaultValueExist) {
+            setRating(currentValue);
+            setTemporaryRating(currentValue);
+        }
+        const ratingDecimalParts = Math.round((currentValue % Math.floor(currentValue)) * 100);
+
+        if (ratingDecimalParts > 0) {
+            setRemainingRating(ratingDecimalParts);
+        }
+
+        if (currentValue < 1) {
+            setRemainingRating(Math.ceil(currentValue * 100));
+        }
+    }, [defaultValue, isDefaultValueExist, value, isControlled, currentValue]);
+
+    const mouseLeaveHandler = () => {
+        setHoveredValue(0);
+        setRegardingPosition(0);
+        setDisableMouseMovie(false);
+        setRating(temporaryRating);
+    };
+
+    const ratingController = (currentRating: number, state: number, blockMouseMovie = true) => {
+        setHoveredValue(currentRating);
+        setRating(0);
+        setTemporaryRating((prev: number) => {
+            if (state !== prev) return state;
+            setHoveredValue(0);
+            setRating(defaultValue);
+            setDisableMouseMovie(blockMouseMovie);
+            return defaultValue;
+        });
+    };
+
+    const getRating = (e: MouseEvent<HTMLDivElement>, currentRating: number) => {
+        if (readonly) return;
+        setRegardingPosition(calculateRegardingPosition(e));
+        const state = regardingPosition === 50 ? +`${currentRating - 1}.${regardingPosition}` : currentRating;
+        if (isControlled) {
+            setDisableMouseMovie(true);
+            setHoveredValue(0);
+            setRating(state);
+            onChange?.(state);
+            return;
+        }
+
+        ratingController(currentRating, state);
+    };
+
+    const mouseEnterHandler = () => {
+        setTemporaryRating(rating);
+    };
+
+    const elementsList = Math.round(count) > 0 ? new Array(count).fill(undefined) : [];
+
+    const PrimitiveValue = (typeof character === 'string' || typeof character === 'number') && character;
+
+    const keyDownHandler = (e: KeyboardEvent<HTMLDivElement>, currentRating: number) => {
+        if (readonly) return;
+
+        if (e.key === 'Enter') {
+            if (isControlled) {
+                setRegardingPosition(100);
+                onChange?.(currentRating);
+
+                return;
+            }
+
+            setRegardingPosition(0);
+            ratingController(currentRating + 1, currentRating, false);
+        }
+    };
+
+    const mouseLeaveHandlerForEveryElement = () => {
+        setDisableMouseMovie(false);
+    };
+
+    return (
+        <div
+            {...restProps}
+            className="rating"
+            onMouseLeave={mouseLeaveHandler}
+            onMouseEnter={mouseEnterHandler}
+            onBlur={() => setDisableMouseMovie(false)}
```

Comment:
You can use an already-defined handler.

```suggestion
            onBlur={mouseLeaveHandlerForEveryElement}
```